### ISSUE TITLE
Notation for out-of-scope variables conflicts with notation for irrelevance

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#3127]
This is SUPER confusing and pops up all the time when working with irrelevant arguments. Example:
```agda
f : {A : Set} -> .A -> A
f = {!!}
```
shows goal type:
```
Goal: ..A → .A
————————————————————————————————————————————————————————————
.A : Set
```
This is especially confusing given that double-dot is also the current notation for (the undocumented) shape-irrelevance.

My proposal would be to print not-in-scope arguments using `{}`:
```
Goal: .A → A
————————————————————————————————————————————————————————————
{A} : Set
```
This would require we rename hidden variables when they are shadowed by a visible variable, something that's currently not necessary. But I think it would increase readability of the goal types quite a bit.

**Comment by andreasabel** [comment]
That would work in the hypotheses display, but we should also have a notation inside types.  My plan was to exchange the `.` for an `@`, but maybe this will now be confusing with attributes.  What about `'a`?
A single quote usually opens a character literal, thus, it will not be parseable (that addresses one of @nad's main concern).

Update: This could be confusing with a shadowed identifier `a'` which would be printed as `'a'` (character literal).
A possibility would be `''a`.

**Comment by andreasabel** [comment]
I implemented the `''` prefix but it does not look good since often identifiers end in `'`.
Another option would be `;`.

**Comment by andreasabel** [comment]
I now switched to ';' which looks acceptable and does not conflict with any parts of the term syntax that the printer produces.
I close this issue, feel free to reopen if you wish to continue bikeshedding.

**Comment by andreasabel** [comment]
The semicolon prefix confuses my reading a lot.  Too noisy.  I am a bit at a loss now.  
Anyone else tried it yet?

**Comment by andreasabel** [comment]
We could reserve an obscure Unicode character to not be legal in a name (test in scope checker!) and use this character as prefix maybe.

**Comment by jespercockx** [comment]
The current situation is unacceptable, the semicolons are way too noisy.

My proposal would be to drop the prefix entirely and print the context entry with curly braces (see my original post). This would remove the noise completely and I think it would still be clear to the user. Are there any arguments against this (other than that it takes some more effort to implement)?

**Comment by nad** [comment]
Perhaps we shouldn't use `{A} : ...`, which suggests that `A` is an implicit argument. (One alternative might be to write `(A) : ...`, another might be to write `A : ... (not in scope)`.) Otherwise I think the original suggestion sounds fine.

**Comment by jespercockx** [comment]
I'm currently trying out the approach without any prefix and while there's still a few things to fix (mostly related to #572) I really have to say ALL THE ERROR MESSAGES LOOK SO MUCH NICER! You will be amazed once I push this change.
